test/libsolidity/semanticTests/array/invalid_encoding_for_storage_byte_array.sol

```diff
@@ -0,0 +1,92 @@
+        delete y;
+    }
+    function abiEncode() public view returns (bytes memory) { return x; }
+    function abiEncodePacked() public view returns (bytes memory) { return abi.encodePacked(x); }
+    function copyToMemory() public view returns (bytes memory m) { m = x; }
+    function indexAccess() public view returns (byte) { return x[0]; }
+    function assignTo() public { x = "def"; }
+    function assignToLong() public { x = "1234567890123456789012345678901234567"; }
+    function copyToStorage() public { y = x; }
+    function copyFromStorageShort() public { y = "abc"; x = y; }
+    function copyFromStorageLong() public { y = "abc"; x = y; }
+    function arrayPop() public { x.pop(); }
+    function arrayPush() public { x.push("t"); }
+    function arrayPushEmpty() public { x.push(); }
+    function del() public { delete x; }
+}
+// ----
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncode() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// abiEncodePacked() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToMemory() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// indexAccess() -> 0x6100000000000000000000000000000000000000000000000000000000000000
+// arrayPushEmpty()
+// arrayPush()
+// x() -> 0x20, 5, 0x6162630074000000000000000000000000000000000000000000000000000000
+// arrayPop()
+// assignToLong()
+// x() -> 0x20, 0x25, 0x3132333435363738393031323334353637383930313233343536373839303132, 0x3334353637000000000000000000000000000000000000000000000000000000
+// assignTo()
+// x() -> 0x20, 3, 0x6465660000000000000000000000000000000000000000000000000000000000
+// copyFromStorageShort()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyFromStorageLong()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// copyToStorage()
+// x() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// y() -> 0x20, 3, 0x6162630000000000000000000000000000000000000000000000000000000000
+// del()
+// x() -> 0x20, 0x00
+// invalidateXLong()
+// x() -> FAILURE
+// abiEncode() -> FAILURE
+// abiEncodePacked() -> FAILURE
+// copyToMemory() -> FAILURE
+// indexAccess() -> FAILURE
+// arrayPushEmpty()
+// arrayPush()
+// x() -> FAILURE
+// arrayPop()
+// assignToLong() -> FAILURE
+// x() -> FAILURE
+// assignTo() -> FAILURE
+// x() -> FAILURE
+// copyFromStorageShort() -> FAILURE
+// x() -> FAILURE
+// copyFromStorageLong() -> FAILURE
+// x() -> FAILURE
+// copyToStorage() -> FAILURE
+// x() -> FAILURE
+// y() -> 0x20, 0x00
+// del() -> FAILURE
+// x() -> FAILURE
+// invalidateXShort()
+// x() -> FAILURE
+// abiEncode() -> FAILURE
+// abiEncodePacked() -> FAILURE
+// copyToMemory() -> FAILURE
+// indexAccess() -> FAILURE
+// arrayPushEmpty()
+// arrayPush()
+// x() -> FAILURE
+// arrayPop()
+// assignToLong()
```

Comment:
This currently does not fail, but I would argue that it should fail to assign something to an invalidly encoded storage array. Basically anything that touches it should fail and the only way to make it right is using inline assembly.

Reply:
I think either is fine here. There's no risk of misinterpreting the data so letting the user just pave over it and forget it would be fine too :)

Reply:
The problem is that we have to delete the old value and that needs interpreting the length.

Reply:
I only meant the case with short array and length > 31. But maybe making a special case of that is not really a good idea. In any case, I think that reverting is valid here too.

Reply:
Even short array with length > 31 - we don't know if the length value is wrong or the fact that it has been encoded in-place. We might have to delete the out-of place slots as well.